### PR TITLE
Add AggWAvg support for rollup()

### LINKED ISSUE #3350: AggWAvg support for rollup()
We currently have AggWSum support for rollup(), but not AggWAvg.

### PR DESCRIPTION
**Pull request by lbooker42**
Will close #3350

Sample groovy code and results:
```
import static io.deephaven.csv.CsvTools.readCsv
import io.deephaven.api.agg.spec.AggSpec
import io.deephaven.api.agg.Aggregation

insurance = readCsv("https://media.githubusercontent.com/media/deephaven/examples/main/Insurance/csv/insurance.csv")

aggList = [Aggregation.of(AggSpec.wavg("age"), "bmi", "expenses")]

insuranceRollup = insurance.rollup(aggList, true, "region", "age")

// Get the overall weighted for manual comparison
insurance_all_wavg = insurance.aggBy(aggList)
insurance_southwest_wavg = insurance.where("region='southwest'").aggBy(aggList)
insurance_southwest_19_wavg = insurance.where("region='southwest'", "age = 19").aggBy(aggList)
insurance_southwest_23_wavg = insurance.where("region='southwest'", "age = 23").aggBy(aggList)
```

We can verify the rollup produces the same values as manually computing a limited part of the table:
image

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java [hunk 1]
```diff
@@ -1288,6 +1289,29 @@ private void reaggregateAvgOperator() {
...
+                // Make a recording operator for the sum of weights column
+                final String sumOfWeightsName = resultName + ROLLUP_SUM_WEIGHTS_COLUMN_ID + ROLLUP_COLUMN_SUFFIX;
+                final ColumnSource<?> sumOfWeightsSource = table.getColumnSource(sumOfWeightsName);
+
+                final DoubleWeightRecordingInternalOperator doubleWeightOperator;
+                doubleWeightOperator = new DoubleWeightRecordingInternalOperator(sumOfWeightsSource.getChunkType());
+                addOperator(doubleWeightOperator, sumOfWeightsSource, sumOfWeightsName);
+
+                final String valueSourceName = pair.input().name();
+                final ColumnSource<?> valueSource = table.getColumnSource(valueSourceName);
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java:1305]:
```suggestion
                final ColumnSource<?> weightedAveragesSource = table.getColumnSource(resultName);
```

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java [hunk 2]
```diff
@@ -1288,6 +1289,29 @@ private void reaggregateAvgOperator() {
...
+
+                // The sum of weights column is directly usable as the weights for the WAvg re-aggregation.
+                final IterativeChunkedAggregationOperator resultOperator = new ChunkedWeightedAverageOperator(
+                        sumOfWeightsSource.getChunkType(), doubleWeightOperator, resultName, true);
+
+                addOperator(resultOperator, valueSource, valueSourceName, sumOfWeightsName);
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java:1311]:
```suggestion
                addOperator(resultOperator, valueSource, resultName, sumOfWeightsName);
```

**Review by rcaudy — COMMENTED** [review]
LGTM. Please confirm you've tested with different input/output column names, different weight and value types, and ticking data sets.

**Comment by lbooker42** [comment]
Tested manually with int/doubles as weights and values, as well as a ticking table:

Groovy code:

```
import static io.deephaven.csv.CsvTools.readCsv
import io.deephaven.api.agg.spec.AggSpec
import io.deephaven.api.agg.Aggregation

insurance = readCsv("https://media.githubusercontent.com/media/deephaven/examples/main/Insurance/csv/insurance.csv")

aggList = [Aggregation.of(AggSpec.wavg("age"), "wavg_bmi=bmi", "wavg_exp=expenses")]

// weight column is int, value columns are double
insuranceRollup = insurance.rollup(aggList, true, "region", "age")

// Get the overall weighted averages and compare manually
insurance_all_wavg = insurance.aggBy(aggList)
insurance_southwest_wavg = insurance.where("region='southwest'").aggBy(aggList)
insurance_southwest_19_wavg = insurance.where("region='southwest'", "age = 19").aggBy(aggList)
insurance_southwest_23_wavg = insurance.where("region='southwest'", "age = 23").aggBy(aggList)

// Recast and repeat

insurance = insurance.update("age=(double)age", "bmi=(int)bmi")

// weight column is double, value columns are int, double
insuranceRollup = insurance.rollup(aggList, true, "region", "age")

// Get the overall weighted averages and compare manually
insurance_all_wavg = insurance.aggBy(aggList)
insurance_southwest_wavg = insurance.where("region='southwest'").aggBy(aggList)
insurance_southwest_19_wavg = insurance.where("region='southwest'", "age = 19").aggBy(aggList)
insurance_southwest_23_wavg = insurance.where("region='southwest'", "age = 23").aggBy(aggList)

/////////////////

t = timeTable("PT00:00:00.001").update("I=i", "J = (int)(i%10)", "K=(long)(epochMillis(Timestamp)/1000)")
h = t.rollup([Aggregation.of(io.deephaven.api.agg.spec.AggSpec.wavg("K"), "I")], true, "J", "K")

h_all = t.aggBy([Aggregation.of(io.deephaven.api.agg.spec.AggSpec.wavg("K"), "I")])
h_0 = t.where("J=0").aggBy([Aggregation.of(io.deephaven.api.agg.spec.AggSpec.wavg("K"), "I")])
h_1 = t.where("J=1").aggBy([Aggregation.of(io.deephaven.api.agg.spec.AggSpec.wavg("K"), "I")])
```

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

Community: https://github.com/deephaven/deephaven.io/issues/3692

**Pull request merged**